recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -20,9 +20,92 @@ processMatch: []
 preInstall:
   requireAtDiscovery: |
       code=$(curl -I http://169.254.169.254/latest/meta-data/instance-id -w %{response_code} -so '/dev/null')
-      if [ $code == "200" ]; then
-        exit 132
+
+      if [ $code != "200" ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/credentials ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/config ]; then
+        exit 1
+      fi
+
+      if ! command -v aws &> /dev/null; then
+        exit 1
+      fi
+
+      callerIdentity=$(aws sts get-caller-identity 2>&1)
+
+      if [ "$(echo "$callerIdentity" | grep AccessDenied | wc -l)" -eq "1" ] ; then
+        exit 1
+      fi
+
+      username=$(echo "$callerIdentity" | sed -ne 's/.*"Arn": ".*\/\(.*\)".*/\1/p')
+
+      echo "username: $username"
```

Comment:
@noahmmcgivern please remove, we should avoid echoing credential type of info now that we've completed the dev work.